internal/infrastructure/mongo/mongodoc/auth_request.go

```diff
@@ -0,0 +1,116 @@
+package mongodoc
+
+import (
+	"time"
+
+	"github.com/caos/oidc/pkg/oidc"
+	"github.com/reearth/reearth-backend/pkg/auth"
+	"github.com/reearth/reearth-backend/pkg/id"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+type AuthRequestDocument struct {
+	ID            string
+	ClientID      string
+	Subject       string
+	Code          string
+	State         string
+	ResponseType  string
+	Scopes        []string
+	Audiences     []string
+	RedirectURI   string
+	Nonce         string
+	CodeChallenge *CodeChallengeDocument
+	CreatedAt     time.Time
+	AuthorizedAt  *time.Time
+}
+
+type CodeChallengeDocument struct {
+	Challenge string
+	Method    string
+}
+
+type AuthRequestConsumer struct {
+	Rows []*auth.Request
+}
+
+func (a *AuthRequestConsumer) Consume(raw bson.Raw) error {
+	if raw == nil {
+		return nil
+	}
+
+	var doc AuthRequestDocument
+	if err := bson.Unmarshal(raw, &doc); err != nil {
+		return err
+	}
+	request, err := doc.Model()
+	if err != nil {
+		return err
+	}
+	a.Rows = append(a.Rows, request)
+	return nil
+}
+
+func NewAuthRequest(req *auth.Request) (*AuthRequestDocument, string) {
+	if req == nil {
+		return nil, ""
+	}
+	reqID := req.GetID()
+	var cc *CodeChallengeDocument
+	if req.GetCodeChallenge() != nil {
+		cc = &CodeChallengeDocument{
+			Challenge: req.GetCodeChallenge().Challenge,
+			Method:    string(req.GetCodeChallenge().Method),
+		}
+	}
+	return &AuthRequestDocument{
+		ID:            reqID,
+		ClientID:      req.GetClientID(),
+		Subject:       req.GetSubject(),
+		Code:          req.GetCode(),
+		State:         req.GetState(),
+		ResponseType:  string(req.GetResponseType()),
+		Scopes:        req.GetScopes(),
+		Audiences:     req.GetAudience(),
+		RedirectURI:   req.GetRedirectURI(),
+		Nonce:         req.GetNonce(),
+		CodeChallenge: cc,
+		CreatedAt:     req.CreatedAt(),
+		AuthorizedAt:  req.AuthorizedAt(),
+	}, reqID
+}
+
+func (d *AuthRequestDocument) Model() (*auth.Request, error) {
+	if d == nil {
+		return nil, nil
+	}
+
+	uuid, err := id.AuthRequestIDFrom(d.ID)
```

Comment:
Just to make sure you're not mistaken, generated IDs of pkg/id is not UUID, but U"L"ID (https://github.com/ulid/spec)